arangod/VocBase/Validators.h

```diff
@@ -53,14 +53,19 @@ struct ValidatorBase {
   explicit ValidatorBase(VPackSlice params);
   virtual ~ValidatorBase() = default;
 
-  bool validate(VPackSlice new_, VPackSlice old_, bool isInsert, VPackOptions const*) const;
+  // Validation function as it should be used in the logical collection or storage engine.
+  Result validate(VPackSlice new_, VPackSlice old_, bool isInsert, VPackOptions const*) const;
```

Comment:
please rename function parameters from `new_` and `old_` to `newSlice` and `oldSlice` or `newDoc` and `oldDoc`.